JacobRintjema/ece452project
# Farm Mode Data Binding + Offline Sync

# Farm Mode Data Binding

**What Was Changed:**
- Connected Firestore to Farm Mode UI
    - Adding new item adds to Firestore
    - Adding existing item updates quantity in Firestore
- SYNC button for offline mode
    - If offline and user tries to update, the data will be stored to SYNC later (when reconnected)
    - User should press SYNC when they re-connect to internet to send data to Firestore

-------

**Testing Steps:**
- Go to offline mode (turn off wifi)
- Add products to table and press update
- Pop will say press SYNC when connected to wifi
    - Table will clear, but the products will be saved
- Turn on wifi
- Press Sync
- Go to Home Mode's inventory -> you should see ur new products

---------------

**Current Issues:**
- Before you press sync/update, if you go to another page everything will reset
    - This is because when you leave and return to a fragment, the onCreateView is triggered and resets all variables
- Random crashes when in Farm Mode (usually switching from Farm Mode -> another mode)

---------------
Added a popup explaining to user that offline mode can't send data to Firestore:

![Screenshot 2023-07-22 at 2 03 31 AM](https://github.com/JacobRintjema/ece452project/assets/55363559/825ca37a-4906-4484-9ed4-b154feb2366e)
